### PR TITLE
Wireless Activity Detector; Power On/Off Activity Detector

### PR DESCRIPTION
**Pull request by slprime**
## Power On/Off Activity Detector

Emit redstone if machine turn on/off

turn-on-off

Cover Gui
image

## Wireless Activity Detector

Item Icon (left - default; right - wireless)
image

In Maschine (left - default; right - wireless)
image

Gui
image

Recipe

| Maintenance (Example) | Activity (New Cover) |
|------------|-----------|
| image | image |

Can be useful in the initial stages when there is a chance of being left without energy

### CODE REVIEW

**Comment by aka-zaratustra** [comment]
It is useful!

**Comment by AbdielKavash** [comment]
Is there a reason why the buttons in the GUI of the wireless version are arranged differently than the basic one?

File: src/main/java/gregtech/common/covers/GT_Cover_DoesWork.java [hunk 1]
```diff
@@ -34,22 +34,25 @@ public boolean isRedstoneSensitive(ForgeDirection side, int aCoverID, int aCover
...
+            } else if (isFlagSet(aCoverVariable, 0x2)) {
+                signal = inverted == (mProgress.getMaxProgress() == 0) ? 0 : 15;
+            } else {
+                int tScale = mProgress.getMaxProgress() / 15;
+
+                if (tScale > 0 && mProgress.hasThingsToDo()) {
+                    signal = inverted ? (15 - mProgress.getProgress() / tScale) : (mProgress.getProgress() / tScale);
                 } else {
-                    aTileEntity.setOutputRedstoneSignal(side, (byte) (aCoverVariable % 2 == 0 ? 0 : 15));
+                    signal = inverted ? 15 : 0;
```
Comment (AbdielKavash) [inline src/main/java/gregtech/common/covers/GT_Cover_DoesWork.java:51]: Would recommend named constants instead of magic values for the flags.

**Review by AbdielKavash — COMMENTED** [review]
Why are the buttons in the GUI of the wireless version arranged differently than in the basic one? This should be made consistent.

**Review by AbdielKavash — COMMENTED** [review]
Rather than "Power", maybe a more correct text would be "Enabled" or "Machine Enabled". "Power" implies that it has something to do with whether the machine is receiving power (EU).

**Comment by YannickMG** [comment]
What problem is this intended to solve?

Could there be a potential overlap with some of the Redstone covers which apparently don't work very well?

File: src/main/java/gregtech/common/covers/GT_Cover_DoesWork.java (hunk 1 above)
Comment (slprime) [reply src/main/java/gregtech/common/covers/GT_Cover_DoesWork.java]: constand added

**Comment by slprime** [comment]
> Why are the buttons in the GUI of the wireless version arranged differently than in the basic one? This should be made consistent.

design changed. top layer with frequency and inverted is basic for all wireless covers

**Comment by slprime** [comment]
> Rather than "Power", maybe a more correct text would be "Enabled" or "Machine Enabled". "Power" implies that it has something to do with whether the machine is receiving power (EU).

change label to "Machine Enabled". (fanks, english not my main language)

**Review by AbdielKavash — APPROVED** [review]

**Review by AbdielKavash — APPROVED** [review]
Good job!

**Pull request merged**